### ISSUE TITLE
Feature: Add `exp` to generated user JWTs

### ISSUE THREAD

**Issue by fullykubed** [edgefarm/vault-plugin-secrets-nats#35]
Great plugin!

We'd like to use it to generate temporary credentials for our NATS clusters.

Unfortunately, the JWTs generated by [this function](https://github.com/edgefarm/vault-plugin-secrets-nats/blob/86002ee5798f7975316dd1437f7eb5d32c7b9e66/paths_issue_user.go#L402) do not have the `exp` set so they are essentially perpetual.

It looks like `exp` support is already present and just requires being added similar to `iss`  [here](https://github.com/edgefarm/vault-plugin-secrets-nats/blob/86002ee5798f7975316dd1437f7eb5d32c7b9e66/paths_issue_user.go#L482).

**Comment by fullykubed** [comment]
Additionally, the expiration time should be checked in the refresh loop so that the stored JWT can be updated before it expires: https://github.com/edgefarm/vault-plugin-secrets-nats/blob/86002ee5798f7975316dd1437f7eb5d32c7b9e66/backend.go#L282.

It seems like it would be good to refresh it when it is >= 50% of the way between `iat` and `exp`.

**Comment by siredmar** [comment]
You are right. Thanks for reporting this! Token expiration is not implemented as this has not been a use case for us yet. 

Since I'm no longer working full time on all these projects, I'll look into this issue when I have some time. 

PRs are very welcome though :)

**Comment by fullykubed** [comment]
@siredmar Thanks for the confirmation. 

We ended up not going with this plugin, otherwise would be happy to add the PR.

For our use case, we didn't need multi-tenancy, just dynamic credentials, and we were able to just use mTLS NATS auth with the Vault PKI backend.
